tests/misc_test.py

```diff
@@ -41,4 +43,39 @@ def test_case_insensitive_dict():
     for k in insensitive_dict.keys():
         del insensitive_dict[randomize_casing(k)]
 
-    assert len(insensitive_dict) == 0
\ No newline at end of file
+    assert len(insensitive_dict) == 0
+
+
+def test_parse_file_uri_scheme():
+    if sys.platform == "win32":
+        p1 = "file://C:/test/doc.txt"
+        assert parse_file_uri(p1) == "C:\\test\\doc.txt"
+        p2 = "file://C:/ProgramData/Cisco/Cisco AnyConnect Secure Mobility Client/aconnect.pac"
+        assert parse_file_uri(
+            p2
+        ) == "C:\\ProgramData\\Cisco\\Cisco AnyConnect Secure Mobility Client\\aconnect.pac"
+
+        p3 = "C:/ProgramData/Cisco/Cisco AnyConnect Secure Mobility Client/aconnect.pac"
+        assert parse_file_uri(
+            p3
+        ) == "C:\\ProgramData\\Cisco\\Cisco AnyConnect Secure Mobility Client\\aconnect.pac"
+
+        p4 = "file:///c:/path/to/the%20file.txt"
+        assert parse_file_uri(
+            p4
+        ) == "C:\\path\\to\\the file.txt"
+
+        # how about a local network location?
+        # e.g: file://hostname/path/to/the%20file.txt
+
+        for p in ["file:///pv/test/doc A.txt", "file:///pv/test/doc%20A.txt"]:
+            assert parse_file_uri(p) == "C:\\pv\\test\\doc A.txt"
```

Comment:
I don't think it should prepended by `c:`. they should start with a single slash (backslash). I would just get rid of scheme `file` and make sure to remve the leading (back)slashes.
In general I recommend just replacing all r`\` with `/` and then doing processing on it. Windows is OK with `/` as separator.

Reply:
line 71 is not a test for network addresses.

currently, i am not sure how to support a network address that is a local file on another machine, so the question/comment.